File: Backend/C#/Backend/Models/CartItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Backend.Validators;
using Newtonsoft.Json;

namespace Backend.Models {
	public class CartItem {
		
		public int Id { get; set; }
		
		[Required]
		[UserExists]
		public int UserId { get; set; }
		
		[JsonIgnore]
		[ForeignKey("UserId")]
		public User User { get; set; }
		
		[Required]
		[ItemExists]
		public int ItemId { get; set; }
		
		[JsonIgnore]
		[ForeignKey("ItemId")]
		public Item Item { get; set; }

		public CartItem() {}

		public CartItem(int userId, int itemId, int id = 0) {
			Id = id;
			UserId = userId;
			ItemId = itemId;
		}

		public override string ToString() {
			return $"{nameof(Id)}: {Id}, {nameof(UserId)}: {UserId}, {nameof(ItemId)}: {ItemId}";
		}
	}
}

File: Backend/java/SpringBoot/src/main/java/nl/bookshop/models/CartItem.java
package nl.bookshop.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import nl.bookshop.validators.ItemExists;
import nl.bookshop.validators.UserExists;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import static nl.bookshop.models.ModelStrings.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = CART_ITEMS)
public class CartItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    @UserExists
    private Integer userId;

    @NotNull
    @ItemExists
    private Integer itemId;

}
